src/pages/metadata-manager.tsx

```diff
@@ -0,0 +1,172 @@
+import { NextPage } from "next";
+import { useAppBridge } from "@saleor/app-sdk/app-bridge";
+import { ConfirmButton, ConfirmButtonTransitionState, makeStyles } from "@saleor/macaw-ui";
+import { SyntheticEvent, useEffect, useState } from "react";
+import Link from "next/link";
+import TextField from "@material-ui/core/TextField";
+import { SALEOR_AUTHORIZATION_BEARER_HEADER, SALEOR_DOMAIN_HEADER } from "@saleor/app-sdk/const";
+import { SettingsApiResponse, SettingsUpdateApiRequest } from "./api/settings";
+
+const useStyles = makeStyles((theme) => ({
+  confirmButton: {
+    marginLeft: "auto",
+  },
+  fieldContainer: {
+    marginBottom: theme.spacing(2),
+  },
+}));
+
+const MetadataManagerPage: NextPage = () => {
+  const { appBridgeState, appBridge } = useAppBridge();
+
+  // Use MakawUI styles to maintain consistency with the rest of the dashboard
+  const classes = useStyles();
+
+  // Transition state is used to visualize submit button state and block form when requests are in-flight
+  const [transitionState, setTransitionState] = useState<ConfirmButtonTransitionState>("loading");
+
+  // State management for two values which will be kept in the metadata
+  const [name, setName] = useState<string>("");
+  const [secret, setSecret] = useState<string>("");
+
+  const handleSubmit = (event: SyntheticEvent) => {
+    event.preventDefault();
+    setTransitionState("loading");
+
+    const newSettings: SettingsUpdateApiRequest = { name, secret };
+
+    fetch("/api/settings", {
+      method: "POST",
+      headers: [
+        ["content-type", "application/json"],
+        [SALEOR_DOMAIN_HEADER, appBridgeState?.domain!],
+        [SALEOR_AUTHORIZATION_BEARER_HEADER, appBridgeState?.token!],
```

Comment:
@krzysztofwolski do you have any idea how to reduce this boilerplate?

Reply:
TBH this example does not check for the bearer header, so this line can be skipped.

Also don't think it's worth abstracting it for example. I think I might pushed too hard with handler helper function.